### PR TITLE
Parse fields as aeson values

### PR DESCRIPTION
**Pull request by cortsf**

### CODE REVIEW

File: Text/CSV/Aeson.hs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+csv = do x <- record `sepEndBy` many1 (oneOf "\n\r")
+         eof
+         return x
+
+record :: Parser Record
+record = (try parseReal <|> try parseInt <|>  try parseNull <|> parseQuotedString <|> parseString) `sepBy` char ','
+
+parseInt :: Parser Field
+parseInt = do
+  many1 digit >>= pure . Number . read
```
Comment (errfrom) [inline Text/CSV/Aeson.hs:54]:
```suggestion
parseInt = Number . read <$> many1 digit
```

File: Text/CSV/Aeson.hs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+parseReal :: Parser Field
+parseReal = do
+  num <- many1 digit 
+  char '.'
+  dec <- many1 digit 
+  pure $ Number (fromFloatDigits (read (num <> "." <> dec) :: Double))
```
Comment (errfrom) [inline Text/CSV/Aeson.hs:61]: Is there a reason why we don't parse the value to a `Scientific` directly?

Before:
```
ghci> parse parseReal "" "1000.123456789123456789"
Right (Number 1000.1234567891235)
```
After:
```
ghci> parse parseReal "" "1000.123456789123456789"
Right (Number 1000.123456789123456789)
```
```suggestion
  pure $ Number $ read $ num <> "." <> dec
```

File: Text/CSV/Aeson.hs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+parseNull :: Parser Field
+parseNull = do
+  string "null" >> pure Null
```
Comment (errfrom) [inline Text/CSV/Aeson.hs:65]:
```suggestion
parseNull = string "null" $> Null
```

File: Text/CSV/Aeson.hs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+parseString :: Parser Field
+parseString = do
+  many (noneOf ",\n\r\"") >>= pure . String . T.pack
```
Comment (errfrom) [inline Text/CSV/Aeson.hs:69]:
```suggestion
parseString = String . T.pack <$> many (noneOf ",\n\r\"")
```

**Review by errfrom — CHANGES_REQUESTED** [review]

**Comment by errfrom** [comment]
I don't know the exact requirements, but perhaps we should consider handling `Bool`s as well?

File: Text/CSV/Aeson.hs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+type Field = Value
+
+-- | A Parsec parser for parsing CSV files
+csv :: Parser CSV
+csv = do x <- record `sepEndBy` many1 (oneOf "\n\r")
+         eof
+         return x
+
+record :: Parser Record
+record = (try parseReal <|> try parseInt <|>  try parseNull <|> parseQuotedString <|> parseString) `sepBy` char ','
```
Comment (errfrom) [inline Text/CSV/Aeson.hs:50]: Redundant space
```suggestion
record = (try parseReal <|> try parseInt <|> try parseNull <|> parseQuotedString <|> parseString) `sepBy` char ','
```

**Review by nalane — CHANGES_REQUESTED** [review]
The types should never be converted from Strings. They need to remain as strings so that they can be spliced into the final JSON

**Comment by nalane** [comment]
This PR is probably no longer needed, right? @cortsf

**Comment by cortsf** [comment]
> This PR is probably no longer needed, right? @cortsf

Yes, we are not using this repo anymore.

**Comment by nalane** [comment]
In that case, I'll go ahead and delete this repo, unless you think it might be good to keep around for any reason

**Comment by cortsf** [comment]
> In that case, I'll go ahead and delete this repo, unless you think it might be good to keep around for any reason

I can't see any reason to keep it. It would be quite rare that somebody could need something like this.

**Comment by cortsf** [comment]
I'll keep a copy just in case (?)

**Pull request closed without merging**